### ISSUE TITLE
CUDA is deprecating text IR support in NVVM, binary IR is required

### ISSUE THREAD

**Issue by elliottslaughter** [zdevito/terra#296]
Thanks to @cinterloper for pointing this out in #288.

> numba/numba#2975
>
> "For the past few CUDA releases, NVVM has supported IR input either in  LLVM 3.4 text IR format or LLVM [later version] bitcode IR format. As of  CUDA 9.2, the bitcode IR format is based on LLVM 5.0."
>
> "There have been warnings in the last few CUDA releases that the text  IR option (which has not been update for some time) is deprecated and  will be dropped in a future CUDA release. See nvvmAddModuleToProgram,  for an example."

If this is true, we will almost certainly want to prioritizing upgrading to LLVM 5.0. Unlike Numba I don't think we're interesting in forking our LLVM or maintaining a separate parser/unparser for the binary IR format.

**Comment by ProfFan** [comment]
According to https://github.com/rust-accel/accel, LLVM 6.0 should be enough for basic CUDA support.

**Comment by ProfFan** [comment]
@elliottslaughter Could you run this on Ubuntu and paste the results?
```
./terra -v tests/cudaatomic.t
```

**Comment by ProfFan** [comment]
Also, https://github.com/AccelerateHS/accelerate-llvm/tree/master/accelerate-llvm-ptx suggests that CUDA 9 only works with LLVM 3.9, which coincides with my observations, and contrary to what the NVIDIA documentation said.

```
// Compiler Build ID: CL-23757830
// Cuda compilation tools, release 9.2, V9.2.64
// Based on LLVM 3.4svn
//
```

**Comment by elliottslaughter** [comment]
Fixed by #299.
